src/java.base/share/classes/sun/nio/ch/SocketChannelImpl.java

```diff
@@ -494,10 +494,7 @@ public int read(ByteBuffer buf) throws IOException {
         }
         long start = SocketReadEvent.timestamp();
         int nbytes = implRead(buf);
-        long duration = SocketReadEvent.timestamp() - start;
-        if (SocketReadEvent.shouldCommit(duration)) {
-            SocketReadEvent.emit(start, duration, nbytes, remoteAddress(), 0);
-        }
+        SocketReadEvent.offer(start, nbytes, remoteAddress(), 0);
```

Comment:
I don't have a strong opinion on shouldCommit+emit vs. offer, both are readable at the use sites. If we go with what you have then we'll need to look into making remoteAddress a stable field.